### ISSUE TITLE
Destructuring argument with default does not work in generators

### ISSUE THREAD

**Issue by monsanto** [6to5/6to5#652]
```
function *foo({bar=0}) {
    console.log(bar)
}

// should print 3, instead prints 0
foo({bar: 3}).next()
```

**Comment by sebmck** [comment]
Nope, this is expected behaviour. Just because you have a destructuring parameter doesn't mean that it automatically defaults to an object.

```javascript
function *foo({ bar = 0 } = {}) {
    console.log(bar)
}
```

**Comment by monsanto** [comment]
I edited that example out pretty quickly, the first case is still wrong.

**Comment by sebmck** [comment]
Variant of #617. Fixed in 3.3.1, thanks!
